src/main/java/teammates/common/datatransfer/attributes/InstructorAttributes.java

```diff
@@ -18,131 +18,49 @@
     public static final String DEFAULT_DISPLAY_NAME = "Instructor";
 
     // Note: be careful when changing these variables as their names are used in *.json files.
+
+    /** Required fields. */
     public String googleId;
     public String courseId;
     public String name;
     public String email;
-    public Boolean isArchived;
+
+    /** Optional fields. */
     public String key;
     public String role;
-    public boolean isDisplayedToStudents;
     public String displayedName;
-
+    public Boolean isArchived;
+    public boolean isDisplayedToStudents;
     public InstructorPrivileges privileges;
 
-    /**
-     * Creates a new instructor with default access level and default displayedName.
-     *
-     * @deprecated only to be used for testing
-     */
-    @Deprecated
-    public InstructorAttributes(String googleId, String courseId, String name, String email) {
```

Comment:
So, I left these parameters as optional. In terms of readability, it's better than constructor with null arguments, especially for tests.